File: baekjoon/10828-스택/정우/10828.py
import sys

n = int(sys.stdin.readline().strip())
arr = []
for _ in range(n):
    line = sys.stdin.readline().strip().split(' ')
    if line[0] == 'push':
        arr.append(line[1])
    elif line[0] == 'top':
        if len(arr) > 0:
            print(arr[-1])
        else:
            print(-1)
    elif line[0] == 'pop':
        if len(arr) > 0:
            print(arr.pop())
        else:
            print(-1)
    elif line[0] == 'size':
        print(len(arr))
    elif line[0] == 'empty':
        print(int(not bool(arr)))

File: baekjoon/2주차/10828-스택/창섭/10828.kt
import java.util.*

fun main() = with(System.`in`.bufferedReader()) {
    val T = readLine().toInt()
    var stack = IntArray(10000)
    var idx = 0
    for (i in 0 until T){
        var command = readLine().split(" ")
        when(command[0]){
            "push"-> stack[idx++] = command[1].toInt()
            "pop"-> {
                if(idx<=0) println(-1)
                else {
                    println(stack[idx-1])
                    idx--
                }
            }
            "size" -> println(idx)
            "empty" -> if (idx!=0) println(0) else println(1)
            "top" -> if (idx!=0) println(stack[idx-1]) else println(-1)
        }


    }
}

